enhancements/microshift/microshift-default-csi-plugin.md

```diff
@@ -0,0 +1,369 @@
+      name: "user-provided-config"
+      volume-group: non_existent_volume_group
+      spare-gb: 10 # in GB
+    socket-name: /run/lvmd/lvmd.sock
+    ```
+**Correct a _malformed_ user-provided ODF-LVM configuration and restart MicroShift**
+
+> After the device owner has logged into the device ...
+
+1. The topolvm-node will validate the config on startup and after detecting a misconfiguration, will enter a
+CrashLoopBackoff state.  
+1. The device owner will observe the topolvm-node crash looping, and access the lvmd container's logs to determine the
+cause of the crash: `oc get logs -n openshift-storage topolvm-node-123yz lvmd`
+1. The device owner will edit the /etc/microshift/lvmd.yaml file to correct the error
+1. The device owner will restart MicroShift: `systemctl restart microshift`
+1. The device owner will validate the system, following workflow **Validate ODF-LVM function**
+
+### API Extensions
+
+**Cluster CRDs**
+
+ODF-LVM represents logical volumes with a CRD called `logicalvolumes.topolvm.cybozu.com`.  This CRD is installed on the 
+cluster automatically by MicroShift during startup. 
+
+### Implementation Details/Notes/Constraints [optional]
+
+#### Default Values
+
+If a user-defined lvmd config is not provided, MicroShift will provide default values, which are geared towards
+[developer](https://github.com/openshift/microshift/blob/main/docs/devenv_rhel8_auto.md) and 
+[production](https://github.com/openshift/microshift/blob/main/docs/rhel4edge_iso.md) deployments. An example default
+configuration would be:
+
+```yaml
+deviceClasses:
+- default: true
+  name: "default"
+  volume-group: "rhel"
+  spare-gb: 10 # in GB
+socket-name: /run/lvmd/lvmd.sock
+```
+
+#### Under the Hood
+
+This workflow ensures that ODF-LVM is reading the most recent version of the lvmd.yaml configuration.  The
+lvmd process reads this once at startup.  Changes to the configuration require that a) the data be pushed to the `lvmd`
+configMap and b) the lvmd process be restarted so that it will pick up the latest configuration.
+
+1. On startup, MicroShift will check for the existence of a file called `/etc/microshift/lvmd.yaml`.  
+   1. If the file exists, MicroShift will read the file into memory
+   1. If the file doesn't exist, MicroShift will assume hardcoded default values.
+1. MicroShift will calculate a SHA256 checksum from the config data. 
+1. MicroShift will generate a ConfigMap to encapsulate the lvmd config data.
+
+    _Example ConfigMap:_
+    ```yaml
+    apiVersion: v1
+    kind: ConfigMap
+    metadata:
+      name: lvmd
+      namespace: openshift-storage
+    data:
+      lvmd.yaml: |
+        device-classes:
+        - name: user-provided-config
+          volume-group: rhel
+          default: true
+        socket-name: /run/lvmd/lvmd.sock
+    ```
+
+1. MicroShift will Create / Update the ConfigMap.
+1. MicroShift will render the topolvm-node daemonset template, with the checksum value passed in as a pod template 
```

Comment:
And the value of the `socket-name` field, right?

Reply:
Ah, yes. Nice catch.